src/solvers/dgsem_p4est/dg_3d.jl

```diff
@@ -21,34 +21,114 @@ function create_cache(mesh::P4estMesh{3}, equations, mortar_l2::LobattoLegendreM
 end
 
 
+#     index_to_start_step_3d(index::Symbol, index_range)
+#
+# Given a symbolic `index` and an `indexrange` (usually `eachnode(dg)`),
+# return an index value to begin a loop and index steps to update during a
+# loop. The resulting indices translate surface indices to volume indices.
+#
+# !!! warning
+#     This assumes that loops using the return values are ordered as
+#
+#     my_i, my_j, my_k = # begin values
+#     for j in ...
+#       for i in ...
+#         # do stuff with (i, j) and (my_i, my_j, my_k)
+#         my_[ijk] += my_[ijk]_step_i
+#       end
+#       my_[ijk] += my_[ijk]_step_j
+#     end
+@inline function index_to_start_step_3d(index::Symbol, index_range)
+  index_begin = first(index_range)
+  index_end   = last(index_range)
+
+  if index === :one
+    return index_begin, 0, 0
+  elseif index === :end
+    return index_end, 0, 0
+  elseif index === :i
+    return index_begin, 1, index_begin - index_end - 1
+  elseif index === :i_backwards
+    return index_end, -1, index_end + 1 - index_begin
+  elseif index === :j
+    return index_begin, 0, 1
+  else # if index === :j_backwards
+    return index_end, 0, -1
+  end
+end
+
+@inline function surface_indices(indices::NTuple{3, Symbol})
```

Comment:
What does this function do? Is there a corresponding function for 2D?

Reply:
I believe this is only necessary in 3D because the volume has three indices and each surface has two. So this function is a way to extract if a surface is looping over `i,j` (analogous to z-direction fixed) or `i,k` (analogous to y-direction fixed) or `j,k` (analogous to x-direction fixed) to draw a comparison to the `TreeMesh3D`

Reply:
No, there isn't since I use a simpler approach in 2D (just checking whether we need to run forward or backwards). I added some comment to explain what this function does.

Reply:
Thanks y'all for the clarification!